converter.py

```diff
@@ -0,0 +1,75 @@
+import tensorflow as tf
+import numpy as np
+
+import sys
+
+# TODO: make sure this takes data from the Accelerate compiler. Right now it
+# has the following assumptions:
+#   1. There are exactly 2 input arrays
+#   2. The shape of both arrays is [100]
+#   3. The representative dataset is random floats in range [0.0, 10.0)
+def representative_data_gen(rng):
+    for _ in range(0, 10):
+        x = rng.random([100], dtype=np.float32) * 10.0
+        y = rng.random([100], dtype=np.float32) * 10.0
```

Comment:
out of curiosity: [0,10) seems like a weird choice, is this something related to how quantisation works, or just arbitrary?

Reply:
I'm not sure if you mean the `for _ in range(0, 10)` or the other two lines, so I'll answer for both:
 * The for loop has an arbitrarily chosen length of 10 iterations. This has to do with how the quantization works: if you do a single run, the code will generate a quantization parameter based on that one run. However, since I'm using random tensors, I want to be sure that the resolution is good. I chose 10 and that seems to work fine with random tensors, at least for small programs. Of course, we should probably update this to work with actual representative data, in which case you would do an iteration for each set of input tensors you pass in.
 * The *10 to multiply the random tensors to put them in range of `[0, 10)` is completely arbitrary. We don't even have an example model to work with, but I had to fill in _something_ and I think I just copied this from another test file :P

Reply:
My comment I meant the second one; an arbitrary choice is fine as long as we know it's temporary.

But since you brought up the second, that's actually just weird. How is this working? Are the names `x` and `y` important here (seems like it can't be), and/or the fact that you are returning two items (possible, but then how does the function relate that to the specific input arrays)? Is that dependent on how the file you are going to execute is set up?

Do you actually need randomly distributed values, like this, or do you only need to specify the minimum and maximum values. Basically is it doing linear quantisation between two limits or is it trying to do something more complicated (which honestly I highly doubt). Can you not just feed it `[10,0,0,0,...]` for example (using your chosen range)?

I would expect that the device is going to generate a single LUT and quantise all inputs to those values, rather than on a per-input basis. If that's the case, knowing that would make our lives a lot easier.

Reply:
Or even just `[min_val, max_val]`, so long as we can feed in the static size information some other way.

Reply:
How the quantization works in a more verbose explanation, addressing your questions at the same time*:
 * The names of the tensors (`x` and `y`) are completely arbitrary and have no impact on the program
 * The fact that the loop is yielding multiple items at once _is_ of high importance. The reason it's returning two values, is because there are two input tensors; now that you mention it, I don't know how to tell TFLite which input is which value in this list, but I think that the order is still set to be the same order as in the old TensorFlow program (i.e. the order in which we generate the inputs).
 * You need to feed _representative data_. The quantization happens on either a per-input or a per-axis** basis, and it also happens for each input/output of each instruction. So in the context of a larger program, it is relevant for the accuracy of the quantization later on in the program which values you feed in at the start of the program. In my understanding, it does some sort of (symbolic?) execution of the program with the data you feed in and uses that to find the range for every input/output of each instruction. As far as I know, the current implementation _does_ do linear quantization, but as a result of propagation through the program, I think we can't just do `[min_range, max_range, ...]`.

*P.S: I learned this information from official TF sources, but I don't exactly remember which ones. I'll link you to [this page about quantization in general](https://www.tensorflow.org/lite/performance/post_training_quantization) and [this one on more technical in-depth details](https://www.tensorflow.org/lite/performance/quantization_spec?hl=sv), but the second one is more aimed at hardware manufacturers, and is more per-instruction based information. I'm not sure if all the information I found is on just those pages, though.
** I'm not sure what the per-axis thing means precisely, but I'm near-certain it does mean greater accuracy for a select few operations. I don't think it's particularly relevant for what we're trying to do, though? It's in the same sources I mentioned before so if you _are_ interested, read the links (I think the second link has more details about this).